compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt

```diff
@@ -402,20 +411,84 @@ internal class ComposeSceneMediator(
     fun setContent(content: @Composable () -> Unit) {
         view.runOnceOnAppeared {
             focusStack?.pushAndFocus(userInputView)
 
             scene.setContent {
                 ProvideComposeSceneMediatorCompositionLocals {
-                    interopContainer.TrackInteropPlacementContainer(
-                        content = content
-                    )
+                    if (configuration.onFocusBehavior == OnFocusBehavior.FocusableAboveKeyboard) {
+                        if (configuration.platformLayers) {
+                            OffsetToFocusedRect(
+                                insets = PlatformInsets(bottom = keyboardOverlapHeight),
+                                getFocusedRect = ::getFocusedRect,
+                                size = scene.size,
+                                animationDuration = if (animateKeyboardOffsetChanges) {
+                                    FOCUS_CHANGE_ANIMATION_DURATION
+                                } else {
+                                    0.seconds
+                                },
+                                animationCompletion = {
+                                    animateKeyboardOffsetChanges = false
+                                }
+                            ) {
+                                interopContainer.TrackInteropPlacementContainer(content = content)
+                            }
+                        } else {
+                            LaunchedEffect(keyboardOverlapHeight) {
+                                scene.invalidatePositionInWindow()
+                            }
+                            LaunchedEffect(animateKeyboardOffsetChanges) {
+                                if (animateKeyboardOffsetChanges) {
+                                    UIView.animateWithDuration(
+                                        duration = FOCUS_CHANGE_ANIMATION_DURATION.toDouble(
+                                            DurationUnit.SECONDS
+                                        ),
+                                        animations = ::updateViewOffset,
+                                        completion = {
+                                            scene.invalidatePositionInWindow()
+                                            animateKeyboardOffsetChanges = false
+                                        }
+                                    )
+                                }
+                            }
+                            OverlayLayout(
+                                modifier = Modifier.onGloballyPositioned {
+                                    updateViewOffset()
+                                }
+                            ) {
+                                interopContainer.TrackInteropPlacementContainer(content = content)
+                            }
+                        }
+                    } else {
+                        interopContainer.TrackInteropPlacementContainer(content = content)
```

Comment:
Please keep only one call of it (subject of refactor). Extraction to a function should resolve it too